rust/kernel/allocator.rs

```diff
@@ -21,6 +21,28 @@ unsafe impl GlobalAlloc for KernelAllocator {
             bindings::kfree(ptr as *const core::ffi::c_void);
         }
     }
+
+    unsafe fn alloc_zeroed(&self, layout: Layout) -> *mut u8 {
+        // `krealloc()` is used instead of `kmalloc()` because the latter is
+        // an inline function and cannot be bound to as a result.
+        unsafe {
```

Comment:
Finn, we are looking to enable the warning on missing `SAFETY` comments before unsafe blocks. Can you add one here?

Reply:
Skipped them, as the already implemented functions where missing them as well. But can do tomorrow

Reply:
Should I add it to the `__rust` functions below as well?

Also think those symbols are technically unsafe, bu we just export symbols, so unsafe informations go missing?